pageserver/src/tenant.rs

```diff
@@ -854,21 +764,24 @@ impl Tenant {
             None
         };
 
-        // Even if there is local metadata it cannot be ahead of the remote one
-        // since we're attaching. Even if we resume interrupted attach remote one
-        // cannot be older than the local one
-        let local_metadata = None;
+        // we can load remote timelines during init, but they are assumed to be so rare that
+        // initialization order is not passed to here.
+        let init_order = None;
+
+        // timeline reattaching requires that we have the metadata locally; it used to be done
+        // in the beginning of timeline_init_and_sync
+        save_metadata(self.conf, &self.tenant_id, &timeline_id, &remote_metadata)
```

Comment:
1. Future readers won't care about how things used to be.
2. They would rather want to know _why_ reattaching requires the metadata file locally.

Please clarify this comment.

Reply:
```suggestion
        // after attach complete, load expects to find the metadata file for each timeline
        save_metadata(self.conf, &self.tenant_id, &timeline_id, &remote_metadata)
```

Reply:
1b5ab4746